src/find.ts

```diff
@@ -39,29 +40,45 @@ const COLLECTION_METHODS = {
  *    -hint {String} An optional index hint to provide to the mongo query
  *    -collation {Object} An optional collation to provide to the mongo query. E.g. { locale: 'en', strength: 2 }. When null, disables the global collation.
  */
-module.exports = async function(collection, params) {
+export interface FindParams extends PaginationParams {
+  query?: Document;
+  limit?: number;
+  fields?: Record<string, number>;
+  collation?: Record<string, any> | null;
+  overrideFields?: Record<string, number>;
+}
+
+const COLLECTION_METHODS = {
+  FIND: 'find',
+  FIND_AS_CURSOR: 'findAsCursor',
+};
+
+export default async function findWithPagination(
+  collection: Collection<Document>,
+  params: FindParams
+): Promise<PaginationResponse<Document>> {
   const removePaginatedFieldInResponse =
     params.fields && !params.fields[params.paginatedField || '_id'];
 
   let response;
   if (params.sortCaseInsensitive) {
     // For case-insensitive sorting, we need to work with an aggregation:
-    response = aggregate(
+    response = await aggregate(
       collection,
       Object.assign({}, params, {
         aggregation: params.query ? [{ $match: params.query }] : [],
       })
     );
   } else {
     // Need to repeat `params.paginatedField` default value ('_id') since it's set in 'sanitizeParams()'
-    params = _.defaults(await sanitizeParams(collection, params), { query: {} });
+    params = _.defaults(await sanitizeParams(collection, params as SanitizeParams), { query: {} });
```

Comment:
Why do we force the use of thet ype SanitizeParams here? Should we just update the type in the parameters of the function?